### PR TITLE
Adding support for csv in MetricsSetup

### PR DESCRIPTION
**Pull request by thomascristanis**
In the current project status only had the support will JMX. So it was implementing the support to generate .csv files in accordance with [metrics library](http://metrics.dropwizard.io/3.1.0/manual/core/#man-core-reporters-csv) already used in the project.

### CODE REVIEW

**Comment by dlaboss** [comment]
Thanks for your contribution! It looks sensible to me, though wish there were some test for it. I see this whole area is lacking in that regard.  I trust you've used it successfully?  @vdogaru can you give it a quick check pls.

**Comment by thomascristanis** [comment]
Yes, I believe it is sensible too. Did the initial tests on my local machine (MacOS), then performed directly on Raspberry Pi. It worked as expected. ([Imagens](https://www.dropbox.com/sh/21b1lm3khn3ppc8/AACkqro0CSfFT43soh82k8t2a?dl=0))
I believe that improvements in file formatting are needed.

Needing help in this test I am at your disposal.

File: utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java [hunk 1]
```diff
@@ -104,15 +104,6 @@ public MetricsSetup startJMXReporter(String jmxDomainName) {
...
 
-        if (pathMetrics == null) { // pathMetrics is NULL
-            pathMetrics = createDefaultDirectory();
-        } else {
-            File directory = new File(pathMetrics);
-            if (!directory.exists() && !directory.mkdirs()) {
-                pathMetrics = createDefaultDirectory();
-            }
-        }
-
```
Comment (dlaboss) [inline utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java]: Is the pathMetrics javadoc still accurate with this change?

**Comment by dlaboss** [comment]
thx for the working confirmation and the images. Can you create an Edgent JIRA to track this enhancement?  If not, I'll create one.

File: utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java (hunk 1 above)
Comment (thomascristanis) [inline utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java]: No, now it's just the argument of this method:

`public MetricsSetup startCSVReporter(String pathMetrics) {

        final CsvReporter reporter = CsvReporter.forRegistry(registry()).formatFor(Locale.US)
                .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS)
                .build(new File(pathMetrics));
        reporter.start(1, TimeUnit.SECONDS);

        return this;
    }`

File: utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java [hunk 2]
```diff
@@ -85,6 +92,25 @@ public MetricsSetup startJMXReporter(String jmxDomainName) {
         reporter.start();
         return this;
     }
+
+    /**
+     * Starts the metric {@code CsvReporter}. If no MBeanServer was set, use the
+     * virtual machine's platform MBeanServer.
+     * 
+     * @param pathMetrics
```
Comment (dlaboss) [inline utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java]: Sorry, what I was trying to ask does this javadoc need to be adjusted now?  Can you still supply "null" or a path to a non-existant file?

Comment (thomascristanis) [inline utils/metrics/src/main/java/org/apache/edgent/metrics/MetricsSetup.java]: I'm making some adjustments to a problem that appeared.
Sorry, I will soon put the set code.

**Comment by thomascristanis** [comment]
I have problems in my fork. I will create another pull request with prefix JIRA

**Comment by vdogaru** [comment]
The change looks good to me.  Thanks!

**Comment by thomascristanis** [comment]
Thanks @vdogaru
Due to problems in the fork in my machine opened a pull request.
[[EDGENT-428] Adding support for csv in MetricsSetup](https://github.com/apache/incubator-edgent/pull/315)

**Pull request closed without merging**